x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_router.tsx

```diff
@@ -95,6 +96,9 @@ export const EngineRouter: React.FC = () => {
       <Route path={ENGINE_DOCUMENTS_PATH}>
         <Documents engineBreadcrumb={engineBreadcrumb} />
       </Route>
+      <Route path={ENGINE_RELEVANCE_TUNING_PATH}>
+        <RelevanceTuning engineBreadcrumb={engineBreadcrumb} />
+      </Route>
```

Comment:
We should be wrapping our routes in ability checks

```suggestion
      {canManageEngineRelevanceTuning && (
        <Route path={ENGINE_RELEVANCE_TUNING_PATH}>
          <RelevanceTuning engineBreadcrumb={engineBreadcrumb} />
        </Route>
      )}
```

Reply:
I just pinged Byron about this in Slack. He was experiencing an issue where if you wrapped a Route in a conditional, it wasn't properly terminating the `Switch`. Waiting to hear what he has to say about that.

Reply:
Ah, he mentioned they switched the path so there was no conflict. Wondering though if we'll run into conflicts with this pattern in the future.

Reply:
Meaning, I wonder if the ability check should be inside of the `Route`, rather than outside the `Route`.

Reply:
I.e., something like:
```
<Route path={ENGINE_ANALYTICS_PATH}>
        {canViewEngineAnalytics ? (
          <AnalyticsRouter engineBreadcrumb={engineBreadcrumb} />
        ) : (
          <NotFound product={APP_SEARCH_PLUGIN} />
        )}
      </Route>
```

Reply:
Meaning, regardless of whether a user has the ability to view a particular page or not, you still want the routing to be the same, you just want to show a different view.

The risk you run otherwise is the switch falling through to a different, unintended view.

Additionally, Byron was also experiencing issues with React Router where having it nested in a conditional didn't appear to actually terminate the `Switch`, it continued to run code from another Route view as wel.

Reply:
I'll add the ability check here, but might be worth circling back to address this.

Reply:
9b245e43c2d8b23bafa307a3b84b30bc7ee55a13

Reply:
Ya let's definitely take a look post-MVP - I'm inclined to leave as-is / 1:1 from the standalone UI for now if it's not actively causing bugs